pkg/agent/agent.go

```diff
@@ -725,13 +725,13 @@ func (i *Initializer) getNodeMTU(localIntf *net.Interface) (int, error) {
 		} else if i.networkConfig.TunnelType == ovsconfig.GRETunnel {
 			mtu -= config.GREOverhead
 		}
+		if i.nodeConfig.NodeIPAddr.IP.To16() != nil {
```

Comment:
You should not use ` IP.To16()` to check if it is an IPv6 address, for IPv4.To16() also not nil.

Reply:
Make sense. Use `To4() == nil` instead.